From 8ac73c2e63af46903169fbde84e46c6cdb3acd44 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Wed, 7 Mar 2018 18:41:21 +0800
Subject: [PATCH 3465/5242] MLK-17735 ARM: imx: pm-rpmsg: ensure the pm is
 late suspended and early resumed

commit  802558870d88c2028c509b129ba809157396716e from
https://source.codeaurora.org/external/imx/linux-imx.git

Since some drivers using rpmsg io as wakeup source enable the wakeup
in suspend stage, then it has to ensure pm rpmsg driver pm sleep is
late suspended and early resumed, otherwise M4 will wakeup A core
directly even if there has no wakeup signal.

Reviewed-by: Robin Gong<yibin.gong@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/pm-rpmsg.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/pm-rpmsg.c b/arch/arm/mach-imx/pm-rpmsg.c
index d2a64df..3bfd6f9 100644
--- a/arch/arm/mach-imx/pm-rpmsg.c
+++ b/arch/arm/mach-imx/pm-rpmsg.c
@@ -314,8 +314,10 @@ static int pm_heartbeat_probe(struct platform_device *pdev)
 };
 MODULE_DEVICE_TABLE(of, pm_heartbeat_id);
 
-static SIMPLE_DEV_PM_OPS(pm_heartbeat_ops, pm_heartbeat_suspend,
-			 pm_heartbeat_resume);
+static const struct dev_pm_ops pm_heartbeat_ops = {
+	 SET_LATE_SYSTEM_SLEEP_PM_OPS(pm_heartbeat_suspend,
+				      pm_heartbeat_resume)
+};
 
 static struct platform_driver pm_heartbeat_driver = {
 	.driver = {
-- 
1.7.9.5

